cogeo_mosaic/backends/base.py

```diff
@@ -18,18 +19,57 @@
 
 from cogeo_mosaic.backends.utils import find_quadkeys, get_hash
 from cogeo_mosaic.cache import cache_config
-from cogeo_mosaic.errors import NoAssetFoundError
+from cogeo_mosaic.errors import NoAssetFoundError, UnsupportedOperation
 from cogeo_mosaic.models import Info, Metadata
 from cogeo_mosaic.mosaic import MosaicJSON
 from cogeo_mosaic.utils import bbox_union
 
 
+def update_zoom_and_bounds(instance, attribute, value):
+    """Update Zooms and Bounds attribute when mosaic_def is set."""
+    instance.minzoom = value.minzoom
+    instance.maxzoom = value.maxzoom
+    instance.bounds = value.bounds
+    return value
+
+
+def mode_validator(instance, attribute, value):
+    """Validate mode."""
+    if value not in instance._available_modes:
+        raise ValueError(
+            f"Invalid '{value}' mode. MUST be one of {instance._available_modes}."
+        )
+
+    if "r" in value:
+        instance._readable = True
+
+    elif "w" in value:
+        instance._writable = True
+
+    if "+" in value:
+        instance._writable = True
+
+
 @attr.s
 class BaseBackend(BaseReader):
-    """Base Class for cogeo-mosaic backend storage."""
+    """Base Class for cogeo-mosaic backend storage.
+
+    Attributes:
+        path (str): mosaic path.
+        mode (str, optional): Access mode. Available modes are: r (read-only), r+ (read and write), w (write). Defaults to read-only `r`.
+        reader (rio_tiler.io.BaseReader): Dataset reader. Defaults to `rio_tiler.io.COGReader`.
+        reader_options (dict): Options to forward to the reader config.
+        backend_options (dict): Global backend options.
+        tms (morecantile.TileMatrixSet, optional): TileMatrixSet grid definition. **READ ONLY attribute**. Defaults to `WebMercatorQuad`.
```

Comment:
Why does this need to be read-only?

Reply:
because mosaicjson doesn't support other projection than mercator for now

Reply:
Ok, why does minzoom, maxzoom need to be readonly?

Reply:
because they get passed from the mosaicjson